### PR TITLE
osd/scrub: improving scrub logs

### PR DESCRIPTION
**Pull request by ronen-fr**
By:
- mainly: reducing the number of log lines;
- fixing the fmtlib formatting of utime_t, and
  creating an optional shorter utime_t format;
- clarifying some log messages;

### CODE REVIEW

**Comment by amathuria** [comment]
@ronen-fr  one of the commits was not signed.

**Comment by ronen-fr** [comment]
> @ronen-fr one of the commits was not signed.

fixed

File: src/osd/scrubber/osd_scrub_sched.cc [hunk 1]
```diff
@@ -204,10 +204,10 @@ void ScrubQueue::register_with_osd(ScrubJobRef scrub_job,
...
   }
 
-  dout(10) << "pg(" << scrub_job->pgid << ") sched-state changed from "
-	   << qu_state_text(state_at_entry) << " to "
-	   << qu_state_text(scrub_job->state)
-	   << " at: " << scrub_job->schedule.scheduled_at << dendl;
+  dout(10) << fmt::format(
+		"pg[{}] sched-state changed from <{:.14}> to <{:.14}>",
+		scrub_job->pgid, state_at_entry, scrub_job->state.load())
+	   << dendl;
```
Comment (amathuria) [inline src/osd/scrubber/osd_scrub_sched.cc:210]: Removal of `scrub_job->schedule.scheduled_at` is intentional here? Just checking.

File: src/osd/PG.cc [hunk 2]
```diff
@@ -1705,10 +1703,10 @@ void PG::reschedule_scrub()
 void PG::on_primary_status_change(bool was_primary, bool now_primary)
 {
   // make sure we have a working scrubber when becoming a primary
-
   if (was_primary != now_primary) {
     ceph_assert(m_scrubber);
-    m_scrubber->on_primary_change(m_planned_scrub);
+    m_scrubber->on_primary_change(
+      "PG::on_primary_status_change", m_planned_scrub);
```
[the remark this answers is not in the document]
Comment (ronen-fr) [reply src/osd/PG.cc:1709]: You are right. Originally I thought there was another on_primary_status_change() out there.
Fixed.

File: src/osd/scrubber/osd_scrub_sched.cc (hunk 1 above)
Comment (ronen-fr) [reply src/osd/scrubber/osd_scrub_sched.cc:210]: don't think so... restored. Thanks!

File: src/osd/PG.cc [hunk 3]
```diff
@@ -1742,10 +1739,10 @@ void PG::on_new_interval()
 
   assert(m_scrubber);
   // log some scrub data before we react to the interval
-  dout(20) << __func__ << (is_scrub_queued_or_active() ? " scrubbing " : " ")
+  dout(30) << __func__ << (is_scrub_queued_or_active() ? " scrubbing " : " ")
            << "flags: " << m_planned_scrub << dendl;
 
-  m_scrubber->on_maybe_registration_change(m_planned_scrub);
+  m_scrubber->on_primary_change("PG::on_new_interval", m_planned_scrub);
```
Comment (amathuria) [inline src/osd/PG.cc:1745]: __func__ here as well. Sorry missed it the first time!

Comment (ronen-fr) [reply src/osd/PG.cc]: Fixed. Thanks.

**Review by amathuria — APPROVED** [review]
LGTM

**Comment by ronen-fr** [comment]
Will be merging based on http://pulpito.front.sepia.ceph.com/?branch=wip-rf-delog1
There are multiple dead jobs - all the RHEL ones. Not related.

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by ronen-fr** [comment]
jenkins test make check

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by lxbsz** [comment]
It seems this PR introduced the following error:

```
[ 76%] Built target librbd
[ 76%] Building CXX object src/osd/CMakeFiles/osd.dir/recovery_types.cc.o
/data/ceph/src/osd/scrubber/osd_scrub_sched.cc: In member function ‘ScrubQueue::scrub_schedule_t ScrubQueue::adjust_target_time(const ScrubQueue::sched_params_t&) const’:
/data/ceph/src/osd/scrubber/osd_scrub_sched.cc:549:45: error: ‘group_digits’ is not a member of ‘fmt’
  549 |                   times.proposed_time, fmt::group_digits(times.min_interval),
      |                                             ^~~~~~~~~~~~
/data/ceph/src/osd/scrubber/osd_scrub_sched.cc:550:24: error: ‘group_digits’ is not a member of ‘fmt’
  550 |                   fmt::group_digits(conf()->osd_scrub_min_interval),
      |                        ^~~~~~~~~~~~
/data/ceph/src/osd/scrubber/osd_scrub_sched.cc:551:24: error: ‘group_digits’ is not a member of ‘fmt’
  551 |                   fmt::group_digits(times.max_interval),
      |                        ^~~~~~~~~~~~
/data/ceph/src/osd/scrubber/osd_scrub_sched.cc:552:24: error: ‘group_digits’ is not a member of ‘fmt’
  552 |                   fmt::group_digits(conf()->osd_scrub_max_interval),
      |                        ^~~~~~~~~~~~
[ 76%] Built target rbd-mirror
[ 76%] Building CXX object src/osd/CMakeFiles/osd.dir/MissingLoc.cc.o
make[2]: *** [src/osd/CMakeFiles/osd.dir/build.make:238: src/osd/CMakeFiles/osd.dir/scrubber/osd_scrub_sched.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 76%] Built target rbd-nbd
```

**Comment by ljflores** [comment]
@ronen-fr ^

**Comment by ronen-fr** [comment]
@lxbsz, @ljflores: that's not a problem in the original PR (considering it was merged in August, and Ceph was built at least hundreds of times since then), but an fmtlib version issue.
Are you compiling in C++20 mode?
@tchaikov - any idea?

**Comment by lxbsz** [comment]
I did nothing and just pulled the latest ceph repo. And the following is my g++ version:

```
$ g++ --version
g++ (GCC) 11.3.1 20220421 (Red Hat 11.3.1-2)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

The os version:

```
$ cat /etc/os-release 
NAME=Fedora
VERSION="34 (Workstation Edition)"
ID=fedora
VERSION_ID=34
VERSION_CODENAME=""
PLATFORM_ID="platform:f34"
PRETTY_NAME="Fedora 34 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:34"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/34/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=34
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=34
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```

**Comment by lxbsz** [comment]
By reverting this PR it worked for me.

**Comment by ronen-fr** [comment]
@lxbsz: reverting the PR is not the right solution.
fmt::group_digits() was introduced into fmtlib 8.1.
Current Ceph has fmtlib at [fmt @ b6f4cea](https://github.com/ceph/fmt/tree/b6f4ceaed0a0a24ccf575fab6c56dd50ccf6f1a9)
which is 8.1.1.
Did you refresh your submodules?

**Comment by lxbsz** [comment]
> @lxbsz: reverting the PR is not the right solution. 

Yeah, just to verify which commits have introduced this and to unblock my work.

> fmt::group_digits() was introduced into fmtlib 8.1. Current Ceph has fmtlib at [fmt @ b6f4cea](https://github.com/ceph/fmt/tree/b6f4ceaed0a0a24ccf575fab6c56dd50ccf6f1a9) which is 8.1.1. Did you refresh your submodules?

Yeah, my ceph repo was just pulled from github and fresh. So all the submodules were also new too. I didn't dig it further what has happened about the `fmt`.

For another VM i didn't hit this issue:

```
# g++ --version
g++ (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

# cat /etc/os-release 
NAME="Red Hat Enterprise Linux"
VERSION="8.6 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.6 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.6
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.6"

```

**Comment by theanalyst** [comment]
Seeing the same issue with centos 8 stream, can't build the latest master anymore with this on. fmt submodule is at 8.1.1

**Comment by ronen-fr** [comment]
@theanalyst - thanks. I'll try to take a look next week.

**Pull request merged**